velox/functions/prestosql/registration/ArrayFunctionsRegistration.cpp

```diff
@@ -212,96 +201,42 @@ void registerArrayFunctions(const std::string& prefix) {
       Array<Generic<T1>>,
       Array<Array<Generic<T1>>>>({prefix + "flatten"});
 
-  registerArrayRemoveFunctions<int8_t>(prefix);
-  registerArrayRemoveFunctions<int16_t>(prefix);
-  registerArrayRemoveFunctions<int32_t>(prefix);
-  registerArrayRemoveFunctions<int64_t>(prefix);
-  registerArrayRemoveFunctions<int128_t>(prefix);
-  registerArrayRemoveFunctions<float>(prefix);
-  registerArrayRemoveFunctions<double>(prefix);
-  registerArrayRemoveFunctions<bool>(prefix);
-  registerArrayRemoveFunctions<Timestamp>(prefix);
-  registerArrayRemoveFunctions<Date>(prefix);
-  registerArrayRemoveFunctions<Varbinary>(prefix);
-  registerArrayRemoveFunctions<Generic<T1>>(prefix);
+  REGISTER_SCALAR_FUNCTIONS_WITHOUT_VARCHAR(
+      registerArrayRemoveFunctions, prefix);
   registerFunction<
       ArrayRemoveFunctionString,
```

Comment:
Do you know the reason why we need to implement a separate function specifically for the VARCHAR type?

Reply:
The StringWriter push_back API uses `copy_from` which is a deep copy, usually in the simple function case, it should be a reference copy.

Reply:
So we use setNoCopy instead of push_back for VARCHAR